File: dotnet/MarkLogic.Client.Tests/TestServiceTests.cs
using System;
using System.Linq;
using Xunit;
using Xunit.Abstractions;

namespace MarkLogic.Client.Tests
{
    public class TestServiceTest : IClassFixture<DatabaseClientFixture>
    {
        private readonly DatabaseClientFixture _dbClientFixture;
        private readonly ITestOutputHelper _output;

        public TestServiceTest(DatabaseClientFixture dbClientFixture, ITestOutputHelper output)
        {
            _dbClientFixture = dbClientFixture;
            _output = output;
        }

        private IDatabaseClient DbClient => _dbClientFixture.DbClient;

        [Fact]
        public async void TestReturnNone()
        {
            await TestService.Create(DbClient).returnNone();
        }

        [Fact]
        public async void TestReturnMultipleAtomic()
        {
            var value1 = "the quick brown fox";
            var value2 = 1234;
            var value3 = DateTime.Now;
            var response = await TestService.Create(DbClient).returnMultipleAtomic(value1, value2, value3);
            _output.WriteLine(response);

            var results = response.Split("\n");
            Assert.Equal(3, results.Length);
            Assert.Equal(value1, results[0]);
            Assert.Equal(value2.ToString(), results[1]);
            Assert.Equal(value3.ToISO8601_3Decimals(), results[2]);
        }

        [Fact]
        public async void TestReturnMultiValue()
        {
            var values = new[] { 1, 2, 3, 4, 5, 6, 7, 8, 9, 10 };
            var results = await TestService.Create(DbClient).returnMultiValue(values);
            results.ToList().ForEach(r => _output.WriteLine(r.ToString()));

            Assert.Equal(values, results);
        }

        [Fact]
        public async void TestReturnDateTime()
        {
            var value = DateTime.Now;
            var result = await TestService.Create(DbClient).returnDateTime(value);
            _output.WriteLine(result.ToISO8601_3Decimals());

            Assert.Equal(value.ToISO8601_3Decimals(), result.ToISO8601_3Decimals());
        }
    }
}

File: src/test/java/com/ds/test/TestServiceTests.java
package com.ds.test;

import java.io.*;
import java.util.*;
import java.util.stream.*;
import java.time.*;

import org.junit.jupiter.api.*;
import com.marklogic.client.*;
import com.ds.TestBase;
import com.ds.test.TestService;


public class TestServiceTests extends TestBase {
  private static DatabaseClient dbClient;
  private static TestService testService;

  @BeforeAll
  public static void setUp() {
    dbClient = getClient();
    testService = TestService.on(dbClient);
  }
  
  @AfterAll
  public static void tearDown() {
    dbClient.release();
  }

  @Test
  public void testReturnNone() throws IOException {
    testService.returnNone();
  }

  @Test
  public void testReturnMultipleAtomic() throws IOException {
    String response = testService.returnMultipleAtomic("the quick brown fox", 1234, LocalDateTime.now());
    System.out.println(response);
  }

  @Test
  public void testReturnMultiValue() throws IOException {
    List<Integer> values = Arrays.asList(1, 2, 3, 4, 5, 6, 7, 8, 9, 10);
    Stream<Integer> response = testService.returnMultiValue(values.stream());
    List<Integer> results = response.collect(Collectors.toList());
    results.forEach(System.out::println);
  }

  @Test
  public void testReturnDateTime() throws IOException {
    LocalDateTime parameter = LocalDateTime.now();
    LocalDateTime response = testService.returnDateTime(parameter);
    System.out.println(String.format("Parameter: %s\nResponse: %s", parameter, response));
  }
}